checks/results.w

```diff
@@ -56,7 +56,10 @@ pub class Results {
   pub inflight latest(checkid: str): CheckResult? {
     let key = this.makeLatestKey(checkid);
     let s = this.bucket.tryGetJson(key);
-    return CheckResult.fromJson(s);
+    if let s = s {
+      return CheckResult.fromJson(s);
+    }
```

Comment:
Yeah - this feels like is a type checking bug in the compiler to me. If `s` is `Json?`, and `fromJson` only accepts `Json`, then the optional value shouldn't be accepted, I think.

(Struct.tryFromJson also sounds like a nice idea 👍 )

Reply:
Yea so first Im an idiot we already have `Struct.tryFromJson` 😁 and you are right this is strange, since `fromJson` expects `Json` so probably as you said some bug in type checking.

Reply:
Im able to reproduce with just:
```js
let f = (): Json? => {
  return nil;
};


struct Foo {
  name: str;
}

Foo.fromJson(f());
```

Going to open an issue

Reply:
Actually simpler example that compiles
```js
let f = (): Json? => {
  return nil;
};

let y = (x: Json) => {};

y(f());
```